test/routes/api/user/user.test.ts

```diff
@@ -0,0 +1,159 @@
+import { it, describe, beforeEach, afterEach, before } from 'node:test'
+import assert from 'node:assert'
+import { build } from '../../../helper.js'
+import { FastifyInstance } from 'fastify'
+
+async function createUser (app: FastifyInstance, userData: Partial<{ username: string; password: string }>) {
+  const [id] = await app.knex('users').insert(userData)
+  return id
+}
+
+async function deleteUser (app: FastifyInstance, username: string) {
+  await app.knex('users').delete().where({ username })
+}
+
+async function updatePasswordWithLoginInjection (app: FastifyInstance, username: string, payload: { currentPassword: string; newPassword: string }) {
+  return await app.injectWithLogin(username, {
+    method: 'PUT',
+    url: '/api/user/update-password',
+    payload
+  })
+}
+
+describe('User API', () => {
+  // Hashed value of `Password123$`
+  const Password123$ = 'ff57faf149a2bcab41bf7ecbbc8ce491.3ce6b34ea3edb3f0a09f811440885bfeda612832c04bfddc9d4b906019d97fa0'
+  let app: FastifyInstance
+
+  before(async () => {
+    const app = await build()
+    const users = ['random-user-0', 'random-user-1', 'random-user-2', 'random-user-3', 'random-user-4', 'random-user-5']
+
+    for (const user of users) {
+      await createUser(app, {
+        username: user,
+        password: Password123$
+      })
+    }
+
+    await app.close()
+  })
+
+  beforeEach(async () => {
+    app = await build()
+  })
+
+  afterEach(async () => {
+    await app.close()
+  })
+
+  it('Should update the password successfully', async () => {
+    await createUser(app, { username: 'random-user-0', password: Password123$ })
+    const res = await updatePasswordWithLoginInjection(app, 'random-user-0', {
+      currentPassword: 'Password123$',
+      newPassword: 'NewPassword123$'
+    })
+
+    assert.strictEqual(res.statusCode, 200)
+    assert.deepStrictEqual(JSON.parse(res.payload), { message: 'Password updated successfully' })
+
+    await deleteUser(app, 'random-user-0')
```

Comment:
Why do you delete the users at the end of each test?
Just clean all of them in `after` hook when you are done.

Reply:
Tests should be run in isolation as much as possible, shared dependencies across preparation hooks are hell.

But if that's a style you prefer, I can use hooks

Reply:
I told you that because you were creating all the users in the `before` hook, but I agree with you.